### ISSUE TITLE
Using "cheaper"/faster cloud computing

### ISSUE THREAD

**Issue by flobotics** [alicevision/meshroom#1671]
Hi,
as everything (number of pictures, increased images size, etc. ) went on, also the prices for power, wouldn't it be good to use more cloud computing ?

If i run a personal computer for e.g. 20 hours or longer, or run a cloud PC for e.g. 20 minutes (even if i need 2 hours for uploading my images) , the costs for the cloud computing is cheaper and it is much faster, so a win-win. The bad thing is, this works for e.g. rendering some blender stuff, but not with meshroom, because it seemed not using much GPU power, mostly CPU (from meshroom-statistics view it only uses CPU)

Are there some projects to use more GPU (if reasonable TPU) inside meshroom ?  Perhaps github-copilot  ( https://copilot.github.com/ ) which is very cool, could help there ? 

Also i wanna say, very good work, thanks alot for meshroom. Go on like you think

**Comment by natowi** [comment]
You can run Meshroom in the cloud, but for the setup you are mostly on your own for now, as it is quite time consuming to document. 

The easiest solution is to run meshroom from cli in a notebook with gpu or on a cloud vm:
https://github.com/alicevision/meshroom/wiki/Meshroom-in-Google-Colab-(cloud)
https://github.com/alicevision/meshroom/wiki/Meshroom-on-AWS

Meshroom supports a few renderfarm submitters, but they are not trivial to setup for the casual user.

**Comment by flobotics** [comment]
yes, you can run meshroom on cloud (even with GUI Installing an X Windows System desktop environment inside https://cloud.google.com/architecture/chrome-desktop-remote-on-compute-engine) .

The installation of meshroom prebuild package is, download, unpack ,run.  Simple.

But the usage of the GPUs is not simple.

Perhaps a "nvidia-smi" output of the build-machine would be nice to even match the cuda-10 and  compute capability 3.0 to 7.5 and the GPU type.

I tried:

1x A100 GPU  with cuda 11.3   , nearly no GPU is used (till sfm)
4x T4 GPUs with cuda 11.??? ,  no GPU, only CPU (till sfm)
1x RTX 3080 laptop GPU cuda 11.6, driver-version 511.79 nvidia-smi 511.79   it uses GPU with Feaure-Extraction, Feature-Matching, Structure-from-motion (till sfm)   (but floating with a max of 50% GPU usage, say statistics-view)

Need to make better statistics.

So using cuda-10 is not the only solution :)

**Comment by flobotics** [comment]
hi, most probably you know there is this https://cloud.google.com/free/docs/gcp-free-tier  where you can use google cloud for a limit up to 300 dollars. (there is also one for amazon aws (https://aws.amazon.com/free/?all-free-tier.sort-by=item.additionalFields.SortRank&all-free-tier.sort-order=asc&awsf.Free%20Tier%20Types=*all&awsf.Free%20Tier%20Categories=*all)  and others too.

But the thing i mention this is, as a project like meshroom, known for user-base and continous working on code, i would think asking google (amazon ,etc.) for some free accounts on cloud computing engines with X-GPUs (perhaps X-TPUs) to get the best out of meshroom, which would benefit for both parties.  

The meshroom developers could develop faster because they get faster responses, not waiting hours.  And the cloud-engine providers can hope for users that use their cloud-engines with meshroom-pre-build-vm-instances and pay money. (Users even would save some money if you calculate power-costs against running time, and other stuff (on gce a 64vCPU machine with 4xT4 GPUs cost per hour 0.94 dollar, etc.).


Just a thought of me :)
